activejob/lib/active_job/log_subscriber.rb

```diff
@@ -45,6 +45,34 @@ def enqueue_at(event)
     end
     subscribe_log_level :enqueue_at, :info
 
+    def enqueue_all(event)
+      info do
+        jobs = event.payload[:jobs]
+        adapter = event.payload[:adapter]
+
+        # Currently, `enqueue_all` is implemented only by SidekiqAdapter and it does
+        # not set `successfully_enqueued` for jobs to any value yet, so need to consider nils too.
+        enqueued_jobs = jobs.select { |job| job.successfully_enqueued? || job.instance_variable_get(:@successfully_enqueued).nil? }
```

Comment:
Currently only sidekiq natively supports this feature. But it is not possible to tell yet which jobs were enqueued and which were not, until https://github.com/sidekiq/sidekiq/pull/5766 is released and the SidekiqAdapter is changed to use that change and to set `successfully_enqueued` attribute https://github.com/rails/rails/blob/6c43ebc220428ce9fc9569c2e5df90a38a4fc4e4/activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb#L40-L57, we need to check for `nil`s too.

Reply:
https://github.com/bensheldon/good_job supports it as well FWIW.

Reply:
Yes, I implemented `enqueue_all` in GoodJob but now realize it does not set `job.successfully_enqueued = true` (though I am assigning a `provider_job_id` if successfully enqueued). 

I will make that change.

Reply:
Could we just assume it worked if we can't tell? I'm really uncomfortable with this workaround.

Reply:
I also agree with @byroot. My preference is `nil` does not imply successfully enqueued.

There is [`payload[:enqueued_count]`](https://github.com/bensheldon/rails/blob/2ac4223c87cf71ba4ec3fda011b9212513e7cd5c/activejob/lib/active_job/instrumentation.rb#L10) available, which I think provides enough information along with `payload[:jobs]` to determine how many jobs were enqueued or not enqueued. It makes the logic slightly more complicated to feature detect whether the adapter implements `job.successfully_enqueued = true`, but the presence of _any_ job with a `true` value would imply that it's implemented. e.g.

```ruby
if payload[:enqueued_count] == jobs.size
  # successfully enqueued #{jobs.size}
elsif jobs.any?(&:successfully_enqueued?)
  # break down what is or isn't enqueued
else
  # can't break down, generic message
 end

```

Reply:
Updated with the @bensheldon's approach. 

There was also a need to fix sidekiq adapter to return the actual number for `:enqueued_count`. And a fix for `#perform_all_later` to return the number to be properly set for the `:enqueued_count`.